### PR TITLE
#461 

### PR DESCRIPTION
**Pull request by Rzpeg**
#461 Cancellation support for ITransport, ASB bottleneck cancellation.
---
Rebus is [MIT-licensed](https://opensource.org/licenses/MIT). The code submitted in this pull request needs to carry the MIT license too. By leaving this text in, __I hereby acknowledge that the code submitted in the pull request has the MIT license and can be merged with the Rebus codebase__.

### CODE REVIEW

File: Rebus/Workers/ThreadBased/ThreadWorker.cs [hunk 1]
```diff
@@ -183,14 +187,18 @@ void DoWork(bool onlyRunContinuations = false)
...
             {
-                var message = await _transport.Receive(transactionContext);
+                var message = await _transport.Receive(transactionContext, cToken);
 
                 return new ReceiveResult(message);
             }
+            //catch (TaskCanceledException tex)
+            //{
+            //    return new ReceiveResult(tex, true);
+            //}
```
Comment (mookid8000) [inline Rebus/Workers/ThreadBased/ThreadWorker.cs]: did you mean to leave these lines commented-out? 😁

Comment (Rzpeg) [inline Rebus/Workers/ThreadBased/ThreadWorker.cs]: Yep. I wanted to make a special treatment for cancelation but for now, a simple log warning works :)

Comment (Rzpeg) [inline Rebus/Workers/ThreadBased/ThreadWorker.cs]: Just forgot to remove the lines tbh :smile:

Comment (mookid8000) [inline Rebus/Workers/ThreadBased/ThreadWorker.cs]: fair enough 😁  I'm on vacation right now and I will be back on Thursday next week.... I cannot promise that I will merge your PR before then...I hope that's ok 😄

**Comment by Rzpeg** [comment]
Sure, have a nice vacation :smile:

**Comment by mookid8000** [comment]
Awesome work 👍 thanks! 😄

**Pull request merged**